### PR TITLE
Mapbox-gl-draw: modes property, custom modes extention

### PR DESCRIPTION
**Pull request by diprokon**
@types/mapbox-gl-draw

Modes are static proporty
Modes enum is not static property
Extend DrawCustomMode type
Add constructor options type

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Modes List](https://github.com/mapbox/mapbox-gl-draw/blob/0efdb4d14a0ecae37cfe467334aaa5b2e501f702/index.js#L31), [Modes names enum](https://github.com/mapbox/mapbox-gl-draw/blob/0efdb4d14a0ecae37cfe467334aaa5b2e501f702/src/api.js#L25)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by diprokon** [comment]
Hi @tudorgergely !
Could you, please, check this?
Thank you

File: types/mapbox__mapbox-gl-draw/index.d.ts [hunk 1]
```diff
@@ -159,51 +159,65 @@ declare namespace MapboxDraw {
...
+        [modeKey: string]: DrawCustomMode;
+        draw_line_string: DrawCustomMode;
+        draw_polygon: DrawCustomMode;
+        draw_point: DrawCustomMode;
+        simple_select: DrawCustomMode;
+        direct_select: DrawCustomMode;
+        static: DrawCustomMode;
+    }
+
+    type IMapboxDrawOptions = ConstructorParameters<typeof MapboxDraw>[0];
```
Comment (jakebailey) [inline types/mapbox__mapbox-gl-draw/index.d.ts:214]: It seems like this should be `MapboxDrawOptions`, without the `I` prefix to fit with the rest of this code.

Comment (jakebailey) [reply types/mapbox__mapbox-gl-draw/index.d.ts:214]: Also, this appears to only exist for testing; is this required to add?

File: types/mapbox__mapbox-gl-draw/index.d.ts [hunk 2]
```diff
@@ -159,51 +159,65 @@ declare namespace MapboxDraw {
     }
 
     interface DrawCustomMode<CustomModeState = any, CustomModeOptions = any> {
-        onSetup?(this: DrawCustomModeThis, options: CustomModeOptions): CustomModeState;
+        onSetup?(this: DrawCustomModeThis & this, options: CustomModeOptions): CustomModeState;
```
Comment (jakebailey) [inline types/mapbox__mapbox-gl-draw/index.d.ts:162]: Hm, this is a bit of an awkward pattern, but I can see why you would need it.

**Review by sandersn — APPROVED** [review]

File: types/mapbox__mapbox-gl-draw/index.d.ts (hunk 1 above)
Comment (diprokon) [reply types/mapbox__mapbox-gl-draw/index.d.ts:214]: I wanted to rename it, but it was to late, sorry
It's not 'required', but really nice to have. In tests I add common use case for it

File: types/mapbox__mapbox-gl-draw/index.d.ts (hunk 2 above)
Comment (diprokon) [reply types/mapbox__mapbox-gl-draw/index.d.ts:162]: Yep, it is awkward pattern, but it's really needed

File: types/mapbox__mapbox-gl-draw/index.d.ts (hunk 1 above)
Comment (jakebailey) [reply types/mapbox__mapbox-gl-draw/index.d.ts:214]: Oh, this got merged, I guess I should have clicked the red button for this comment :(

**Pull request merged**